core/src/main/scala/org/apache/spark/SerializableWritable.scala

```diff
@@ -41,7 +40,6 @@ class SerializableWritable[T <: Writable](@transient var t: T) extends Serializa
   private def readObject(in: ObjectInputStream): Unit = Utils.tryOrIOException {
     in.defaultReadObject()
     val ow = new ObjectWritable()
-    ow.setConf(new Configuration())
```

Comment:
I think that this, and SerilaizableWritable more generally, may be a huge source of perf. bottlenecks for short tasks.  A common use of SerializableWritable is in serializing Hadoop Configurations, but it seems kind of crazy to create and discard a new Configuration in order to be able to deserialize the driver-provided conf.  Maybe we can make a substitute for SerializableWritable which only deals with Configuration subclasses and just calls `writeFields()` and `readFields()` directly.  This would sidestep a lot of the performance penalties involved in creating Configuration objects and having them spend tons of time loading defaults.